tests/menus/test_frequently_used_context_menu.py

```diff
@@ -0,0 +1,120 @@
+import platform
+from time import sleep
+
+import pytest
+from selenium.webdriver import Firefox
+
+from modules.browser_object import ContextMenu, Navigation
+from modules.page_object import ExamplePage
+from modules.util import BrowserActions, Utilities
+
+
+@pytest.mark.unstable
+def test_save_page_as(driver: Firefox):
+    """
+    C2637623.1: save page as
+    """
+    try:
+        from pynput.keyboard import Key
+    except ModuleNotFoundError:
+        pytest.skip("Could not load pynput")
+    # create objects
+    context_menu = ContextMenu(driver)
+    driver.get("https://example.com")
+    example_page = ExamplePage(driver)
+    nav = Navigation(driver)
+    util = Utilities()
+    ba = BrowserActions(driver)
+
+    # right click something that is not a hyperlink
+    title_header = example_page.get_element("title-header")
+    context_menu.context_click_element(title_header)
+
+    with driver.context(driver.CONTEXT_CHROME):
+        save_page_as = context_menu.get_context_item("context-menu-save-page-as")
+        save_page_as.click()
+
+    downloads_button = nav.get_download_button()
+
+    # short sleep to ensure menu is shown
+    sleep(0.5)
+
+    # perform key presses to save the file
+    this_platform = platform.system()
+    if this_platform == "Linux":
+        ba.controller.press(Key.alt)
+        ba.controller.press(Key.tab)
+        ba.controller.release(Key.tab)
+        ba.controller.release(Key.alt)
+
+        ba.controller.press(Key.alt)
+        ba.controller.press(Key.tab)
+        ba.controller.release(Key.tab)
+        ba.controller.release(Key.alt)
+
+        ba.key_press_release(Key.tab)
+
+        ba.key_press_release(Key.tab)
+
+    # Press and release the Enter key
+    ba.key_press_release(Key.enter)
+
+    # Wait for the animation to complete
+    with driver.context(driver.CONTEXT_CHROME):
+        nav.wait_for_download_animation_finish(downloads_button)
+
+    # verify and delete downloaded file
+    saved_image_location = util.get_saved_file_path("Example Domain.html")
+
+    util.check_file_path_validility(saved_image_location)
+
+    util.remove_file(saved_image_location)
+
+
+def test_take_screenshot(driver: Firefox):
+    """
+    C2637623.2: take screenshot works from context menu
+    """
+    # create objects
+    context_menu = ContextMenu(driver)
+    driver.get("https://example.com")
+    example_page = ExamplePage(driver)
+
+    # ensure that the screenshot is not present
+    with driver.context(driver.CONTEXT_CHROME):
+        example_page.element_does_not_exist("take-screenshot-box")
```

Comment:
I think I've refactored this function recently and it doesn't require explicit context switching anymore. If not, I think you could do that!

Reply:
```def element_does_not_exist(self, name: str, labels=[]) -> Page:
        """Expect helper: wait until element exists or timeout"""
        original_timeout = self.driver.timeouts.implicit_wait
        self.driver.implicitly_wait(0)
        self.instawait.until_not(
            EC.presence_of_all_elements_located(self.get_selector(name, labels=labels))
        )
        self.driver.implicitly_wait(original_timeout)
        return self``

Sorry if this is a stupid question, where is the chrome context searching taking place in this function?

Reply:
Ok, I'll wait a moment. I'm not getting any changes for this function hm